agent/05_agent_configure.sh

```diff
@@ -245,7 +249,112 @@ function generate_cluster_manifests() {
   ansible-playbook -vvv \
           -e install_path=${SCRIPTDIR}/${INSTALL_CONFIG_PATH} \
           "${SCRIPTDIR}/agent/create-manifests-playbook.yaml"
+}
+
+function add_haproxy_server_lines() {
+  num_servers=${1}
+  type=${2}
+  port=${3}
 
+  for (( n=0; n<${num_servers}; n++ ))
+  do
+    cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+  server ${type}-$n ${type}-$n.${CLUSTER_DOMAIN}:${port} check inter 1s
+EOF
+  done
+}
+
+function enable_load_balancer() {
+  api_ip=${1}
+  load_balancer_ip=${2}
+  if [[ "${AGENT_PLATFORM_TYPE}" == "none" && "${NUM_MASTERS}" > "1" ]]; then
+
+    # setup haproxy as the load balancer
+    if [[ "${IP_STACK}" == "v6" ]]; then
+      # The "wildcard" is different depending on IP stack.
+      # See http://docs.haproxy.org/1.6/configuration.html#4.2-bind
+      export HAPROXY_WILDCARD="[::]"
+    else
+      export HAPROXY_WILDCARD="*"
+    fi
+
+    cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+defaults
+    mode                    tcp
+    log                     global
+    timeout connect         10s
+    timeout client          1m
+    timeout server          1m
+frontend stats
+  bind *:1936
+  mode            http
+  log             global
+  maxconn 10
+  stats enable
+  stats hide-version
+  stats refresh 30s
+  stats show-node
+  stats show-desc Stats for ocp4 cluster
+  stats auth admin:ocp4
+  stats uri /stats
+listen api-server-6443
+  bind ${HAPROXY_WILDCARD}:6443
+  mode tcp
+EOF
+    add_haproxy_server_lines $NUM_MASTERS "master" "6443"
+
+    cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+listen machine-config-server-22623
+  bind ${HAPROXY_WILDCARD}:22623
+  mode tcp
+EOF
+    add_haproxy_server_lines $NUM_MASTERS "master" "22623"
+
+    if [[ "${NUM_WORKERS}" > "0" ]]; then
+      cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+listen ingress-router-443
+  bind ${HAPROXY_WILDCARD}:443
+  mode tcp
+  balance source
+EOF
+      add_haproxy_server_lines $NUM_WORKERS "worker" "443"
+
+      cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+listen ingress-router-80
+  bind ${HAPROXY_WILDCARD}:80
+  mode tcp
+  balance source
+EOF
+      add_haproxy_server_lines $NUM_WORKERS "worker" "80"
+    else
+      cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+listen ingress-router-443
+  bind ${HAPROXY_WILDCARD}:443
+  mode tcp
+  balance source
+EOF
+      add_haproxy_server_lines $NUM_MASTERS "master" "443"
+
+      cat << EOF >> ${WORKING_DIR}/haproxy.cfg
+listen ingress-router-80
+  bind ${HAPROXY_WILDCARD}:80
+  mode tcp
+  balance source
+EOF
+      add_haproxy_server_lines $NUM_MASTERS "master" "80"
+    fi
+
+    sudo firewall-cmd --zone=libvirt --add-port=22623/tcp
+    sudo firewall-cmd --zone=libvirt --add-port=6443/tcp
+    sudo firewall-cmd --zone=libvirt --add-port=443/tcp
```

Comment:
In case of a new firewall port to be opened, it's recommeded to add it here https://github.com/openshift-metal3/dev-scripts/blob/d06b36e96f37feed07fa213207eb471bcff62e8e/02_configure_host.sh#L258 in the step dedicated to setup the firewall (my expectation would be just to find the haproxy port added, the others should be already present)